# UNAH
## Facultad de Ingenieria
## Carrera de Ingenieria en sistemas
### Clase: Inteligencia Artificial

*Prueba*

In [40]:
import pandas as pd
import numpy as np

Segun el siguiente Dataset:

In [41]:
pd_vector = pd.DataFrame( { "num" : range(1,101) } )

In [42]:
pd_vector

num
0     1
1     2
2     3
3     4
4     5
..  ...
95   96
96   97
97   98
98   99
99  100

[100 rows x 1 columns]

Y segun los siguientes metodos de numpy

In [43]:
np.var( pd_vector["num"] ) ## Calcula la varianza

833.25

In [44]:
np.std( pd_vector["num"] ) ## Calcula la desviacion estandard

28.86607004772212

## Responda las siguientes ejercicios

1. Grafique el Boxplot del vector "num"

In [45]:
boxplot = pd_vector.boxplot()

2. Calcule la varianza y desviacion estandard de cada uno de los quartiles

Nota: realice manual la divicion del dataset para obtener cada uno de quartiles, al final obtendra cuatro dataframes

In [14]:
df_split = np.array_split(pd_vector, 4) ##Division de los quartiles en 4 dataframes 


In [27]:
np.std( df_split[0] ) ## desviacion standard primer quartil


num    7.211103
dtype: float64

In [16]:
np.var( df_split[0] ) ## varianza del primer quartil


num    52.0
dtype: float64

In [17]:
np.var( df_split[1] ) ## varianza del segundo quartil


num    52.0
dtype: float64

In [18]:
np.std( df_split[1] ) ## desviacion standard segundo quartil


num    7.211103
dtype: float64

In [19]:
np.var( df_split[2] ) ## varianza del tercer quartil


num    52.0
dtype: float64

In [20]:
np.std( df_split[2] ) ## desviacion standard tercer quartil


num    7.211103
dtype: float64

In [49]:
np.var( df_split[3] ) ## varianza del cuarto quartil


num    52.0
dtype: float64

In [22]:
np.std( df_split[3] ) ## desviacion standard cuarto quartil

num    7.211103
dtype: float64

3. Explique la respuesta que obtuvo en el ejercicio dos, recuerde que se puede apoyar del grafico del ejercicio uno para hacer referencias sobre distribucion en caso de que sea necesario.

## Si nos fijamos la varianza (52) y desviacion estandar (7.21) de los 4 quartiles es la misma, esto debido a que la poblacion de cada quartil esta distribuida de la misma forma, es decir, los 25 elementos tienen una "distancia" entre ellos de 1 unidad, siendo el ultimo elemento 25 unidades mayor que el primer elemento